spec/v1.0-draft/field_types.md

```diff
@@ -46,6 +46,50 @@ _DigestSet (object)_
 > -   `{"sha256": "abcd"}` does not match `{"sha256": "fedb", "sha512": "abcd"}`
 > -   `{"somecoolhash": "abcd"}` uses a non-predefined algorithm
 
+<a id="Reference"></a>
+_Reference (object)_
```

Comment:
I'm still don't think we should allow arbitrary fields, because I want to make sure we don't start conflating references with descriptors of objects (i.e., artifacts or resources) beyond what is needed to identify/find those objects. But your examples got me thinking that it may be useful to come up with a separate Reference sub-type for git repos, and/or other non-hashable resources like VM images. This would make the purpose of the reference clearer while also avoiding potential arbitrary content in the JSON object.

Reply:
+1 on having something git-specific, since it's such a common use case and the SPDX URI is error-prone.

If we don't allow arbitrary field,s where should we stick these other properties? This seems like the natural place to put it. Perhaps there is a dedicated field for that, which allows for arbitrary sub-fields?

Reply:
I want to plug ITE-4 here. I think that'll help with context-specific references as opposed to having it be entirely arbitrary.

Reply:
Ok, so in discussion with other maintainers, this is the approach we came up with. To support reference-specific extensions, we will add an optional `annotations` field to the reference type. In the documentation, we'll mention that (1) the field is intended for any additional information the producer might want to convey about the referenced object, (2) annotations are context specific, so parsers must not fail if there are fields they don't recognize, but there is no expectation that all consumers will know how to use the annotations, and (3) that we are considering supporting context/application-specific standardized annotations objects in the future.

@MarkLodato Does this meet your requirements?

Reply:
@adityasaky Thanks for the reminder about ITE-4. Based on my understanding of ITE-4, I think that's most relevant for references that have a `uri` field set. Not sure what you mean by "context-specific" references? They were never meant to be arbitrary.

Reply:
Sounds good to me!

Reply:
Oh I'm sorry, I wrote my last message in a hurry. I was trying to say that we could use ITE-4 to derive context-specific properties for something as opposed to allowing arbitrary fields, but I see now that doesn't quite work for all use cases.

Separately, I think ITE-4 is still relevant here as an instance of the reference type could apply to an ITE-4 artifact, if I'm not mistaken. And from what I'm reading in the PR, it should work fine, so :+1:!